interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -199,6 +217,37 @@ object TapirAdapter {
     )
   }
 
+  private def encodeTextEventStreamResponse[E, BS](
+    resp: GraphQLResponse[E]
+  )(implicit streamConstructor: StreamConstructor[BS], responseCodec: JsonCodec[ResponseValue]): CalibanBody[BS] = {
+    val response: ZStream[Any, Throwable, ServerSentEvent] = resp.data match {
+      case ObjectValue(fields) =>
+        fields.foldLeft(ZStream.empty: ZStream[Any, Throwable, ServerSentEvent]) { case (_, v) =>
```

Comment:
Won't this handling cause any top-level fields that are not streams to be discarded?

Reply:
Don't think so. I've tested running regular queries with `text/event-stream` as accept header and that works as expected as well. But you mean if a subscription consists of both streams and regular ResponseValues? I can test that as well

Reply:
If my understanding of the spec is correct, we cannot ever get a mix of queries, mutations and/or subscriptions in a single GraphQLRequest. I don't think `GraphQLResponse` currently models this in a correct way so I think this is fine, but maybe changing it to more accurately representing what kind of query was executed would allow us to make less assumptions in the transport layer?

Reply:
We can't get a mix of queries, but isn't it possible to have multiple top-level subscription fields? e.g.,

```graphql
subscription {
  foo {
    A
  }
  bar {
    B
  }
}
```

If I'm reading the code correctly, the code above will not return the `foo` field

Reply:
from the spec: `Subscription operations must have exactly one root field.` 
https://spec.graphql.org/October2021/#sec-Single-root-field

Reply:
Ahh okay, easy then 👍